src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -396,5 +396,63 @@ public static string GenerateLinkingEntityName(string source, string target)
 
             return new(sourceTargetEntityNames[1], sourceTargetEntityNames[2]);
         }
+
+        /// <summary>
+        /// Helper method to extract a hotchocolate field node object with the specified name from all the field node objects belonging to an input type object.  
+        /// </summary>
+        /// <param name="objectFieldNodes">List of field node objects belonging to an input type object</param>
+        /// <param name="fieldName"> Name of the field node object to extract from the list of all field node objects</param>
+        /// <exception cref="ArgumentException"></exception>
+        public static IValueNode GetFieldNodeForGivenFieldName(List<ObjectFieldNode> objectFieldNodes, string fieldName)
+        {
+            ObjectFieldNode? requiredFieldNode = objectFieldNodes.Where(fieldNode => fieldNode.Name.Value.Equals(fieldName)).FirstOrDefault();
+            if (requiredFieldNode != null)
+            {
+                return requiredFieldNode.Value;
+            }
+
+            throw new ArgumentException($"The provided field {fieldName} does not exist.");
+        }
+
+        /// <summary>
+        /// Helper method to determine if the relationship defined between the source entity and a particular target entity is an M:N relationship.
+        /// </summary>
+        /// <param name="sourceEntity">Source entity.</param>
+        /// <param name="relationshipName">Relationship name.</param>
+        /// <returns>true if the relationship between source and target entities has a cardinality of M:N.</returns>
+        public static bool IsMToNRelationship(Entity sourceEntity, string relationshipName)
+        {
+            return sourceEntity.Relationships is not null &&
+                sourceEntity.Relationships.TryGetValue(relationshipName, out EntityRelationship? relationshipInfo) &&
+                !string.IsNullOrWhiteSpace(relationshipInfo.LinkingObject);
+        }
+
+        /// <summary>
+        /// Helper method to get the name of the related entity for a given relationship name.
+        /// </summary>
+        /// <param name="entity">Entity object</param>
+        /// <param name="relationshipName">Name of the relationship</param>
```

Comment:
missing param `entityName`. not a fan of having a parameter that is only used for error messages, but i suppose that's a side effect of the `Entity` object not having a `name` property.

Reply:
Added